tidy3d/plugins/invdes/design.py

```diff
@@ -32,24 +32,63 @@ class AbstractInverseDesign(InvdesBaseModel, abc.ABC):
         description="Task name to use in the objective function when running the ``JaxSimulation``.",
     )
 
+    postprocess: PostprocessOperationType = pd.Field(
+        None,
+        title="Postprocess Object",
+        description="Optional object specifying how to perform weighted sum of ``ModeData`` "
+        "outputs. Can be used instead of passing a generic postprocess function "
+        "to ``Optimizer.run()``.",
+    )
+
     verbose: bool = pd.Field(
         False,
         title="Task Verbosity",
         description="If ``True``, will print the regular output from ``web`` functions.",
     )
 
+    @classmethod
+    def from_function(cls, fn: PostProcessFnType, **kwargs) -> AbstractInverseDesign:
+        """Create an ``InverseDesign`` object from a user-supplied postprocessing function."""
+
+        postprocess = CustomPostprocessOperation.from_function(fn)
+        return cls(postprocess=postprocess, **kwargs)
```

Comment:
Does it really make sense to have `InverseDesign` be able to construct a `CustomPostProcessOperation`? Why does only `postprocess` get this kind of special treatment?
I think there should not be multiple ways of constructing invdes objects from postprocess functions if those ways are functionally identical.

Reply:
So the idea was to make this
```py
postprocess = CustomPostprocessOperation.from_function(fn)
des = InverseDesign(postprocess=postprocess, ...)
```
a bit simpler
```py
des = InverseDesign.from_function(postprocess=postprocess, ...)
```

and save the user from having to know about `CustomPostprocessOperation`.

The reason this gets special treatment I guess is that I think it will be used often enough (especially since we'll recommend users passing functions into `Optimizer.run(f)` to switch to this). and the `CustomPostprocessOperation` basically does nothing but be a class that the `InverseDesign` schema recognizes.

What do you think?